http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterpSpec.scala

```diff
@@ -1482,13 +1482,65 @@ class RuntimeServiceInterpSpec extends AnyFlatSpec with LeonardoTestSuite with T
     res.unsafeRunSync()
   }
 
+  it should "reject a request if requested PD's zone is different from target zone" in isolatedDbTest {
+    val req = PersistentDiskRequest(diskName, Some(DiskSize(500)), None, Map("foo" -> "bar"))
+    val res = for {
+      context <- ctx.ask[AppContext]
+      _ <- makePersistentDisk(Some(req.name), Some(FormattedBy.GCE), None, Some(zone), Some(project)).save()
+      // save a PD with default zone
+      targetZone = ZoneName("europe-west2-c")
+      diskResult <- RuntimeServiceInterp
+        .processPersistentDiskRequest(req,
+                                      targetZone,
+                                      project,
+                                      userInfo,
+                                      serviceAccount,
+                                      FormattedBy.GCE,
+                                      whitelistAuthProvider,
+                                      Config.persistentDiskConfig)
+        .attempt
+    } yield {
+      diskResult shouldBe (Left(
+        BadRequestException(
+          s"requested disk ${diskName.value} is not in the same zone as the requested zone ${targetZone.value}",
```

Comment:
Does the error string need to match exactly?

Reply:
ah yea....I updated the message but forgot to update test